### ISSUE TITLE
Launching anything on Linux is broken

### ISSUE THREAD

**Issue by xddxd** [RPCS3/rpcs3#11446]
After #11445, trying to launch a game or an elf on Linux results in:
![image](https://user-images.githubusercontent.com/1712979/151147824-ef365456-6ebc-4e29-976e-f037f396c908.png)
Log:
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/7940934/RPCS3.log)
Windows is not affected.

**Comment by MSuih** [comment]
`if (!f)` probably needs to be changed to `if (!f || !fs::is_file(filename))`. Errno 21 is EISDIR. Or we can figure out why the overlay is looking for invalid files on linux.

**Comment by Nekotekina** [comment]
Try #11444

**Comment by xddxd** [comment]
Works on that PR.

**Comment by xddxd** [comment]
There is another issue. When there is no Icons directory on first launch, launching something creates this.
![image](https://user-images.githubusercontent.com/1712979/151164043-cf010a7e-7e61-4501-8884-77865bf028cd.png)

**Comment by reece394** [comment]
Same issue on macOS #11371

**Comment by Nekotekina** [comment]
Oh wow, that's a weird one. I hope fixed.

**Comment by reece394** [comment]
Tested all commits that have been built using cirrus CI and the issue still isn't resolved for me yet on macOS. Will have to wait for a new build to be generated to test #11444 fully as they were aborted

**Comment by xddxd** [comment]
![image](https://user-images.githubusercontent.com/1712979/151182593-fb8a2e96-6b1a-4f35-9f67-9a8aed747b2a.png)
Fixed. Clear your Icons directory and wait for a new build I guess.
